homeassistant/bootstrap.py

```diff
@@ -330,77 +342,121 @@ def _get_domains(hass: core.HomeAssistant, config: Dict[str, Any]) -> Set[str]:
+    domains = _get_domains(hass, config)
+    logging_domains = domains & LOGGING_INTEGRATIONS
 
-    async def async_setup_multi_components(domains: Set[str]) -> None:
-        """Set up multiple domains. Log on failure."""
-
-        async def _async_log_pending_setups() -> None:
-            """Periodic log of setups that are pending for longer than LOG_SLOW_STARTUP_INTERVAL."""
-            while True:
-                await asyncio.sleep(LOG_SLOW_STARTUP_INTERVAL)
-                remaining = [domain for domain in domains if domain in setup_started]
-
-                if remaining:
-                    _LOGGER.info(
-                        "Waiting on integrations to complete setup: %s",
-                        ", ".join(remaining),
-                    )
-
-        futures = {
-            domain: hass.async_create_task(async_setup_component(hass, domain, config))
-            for domain in domains
-        }
-        log_task = asyncio.create_task(_async_log_pending_setups())
-        await asyncio.wait(futures.values())
-        log_task.cancel()
-        errors = [domain for domain in domains if futures[domain].exception()]
-        for domain in errors:
-            exception = futures[domain].exception()
-            assert exception is not None
-            _LOGGER.error(
-                "Error setting up integration %s - received exception",
-                domain,
-                exc_info=(type(exception), exception, exception.__traceback__),
-            )
+    # Load logging as soon as possible
+    if logging_domains:
+        _LOGGER.info("Setting up logging: %s", logging_domains)
+        await async_setup_multi_components(hass, logging_domains, config, setup_started)
 
-    domains = _get_domains(hass, config)
+    _LOGGER.info("Domains to be set up: %s", domains)
 
     # Start up debuggers. Start these first in case they want to wait.
     debuggers = domains & DEBUGGER_INTEGRATIONS
+
     if debuggers:
-        _LOGGER.debug("Starting up debuggers %s", debuggers)
-        await async_setup_multi_components(debuggers)
-        domains -= DEBUGGER_INTEGRATIONS
-
-    # Resolve all dependencies of all components so we can find the logging
-    # and integrations that need faster initialization.
-    resolved_domains_task = asyncio.gather(
-        *(loader.async_component_dependencies(hass, domain) for domain in domains),
-        return_exceptions=True,
-    )
+        _LOGGER.debug("Setting up debuggers: %s", debuggers)
+        await async_setup_multi_components(hass, debuggers, config, setup_started)
+
+    # Resolve all dependencies so we know all integrations
+    # that will have to be loaded and start rightaway
+    integrations_to_process = [
+        int_or_exc
+        for int_or_exc in await asyncio.gather(
+            *(loader.async_get_integration(hass, domain) for domain in domains),
+            return_exceptions=True,
+        )
+        if isinstance(int_or_exc, loader.Integration)
+    ]
```

Comment:
When resolving all domains, you need to take into account that the stage 1 domains might not be in the configuration, nevertheless, they have dependencies that need to be resolved.

For example, `alexa` is missing now, while cloud depends on it.

```suggestion
    # Resolve all dependencies so we know all integrations
    # that will have to be loaded and start right away
    all_domains = domains | STAGE_1_INTEGRATIONS
    integrations_to_process = [
        int_or_exc
        for int_or_exc in await asyncio.gather(
            *(loader.async_get_integration(hass, domain) for domain in all_domains),
            return_exceptions=True,
        )
        if isinstance(int_or_exc, loader.Integration)
    ]
```